version/ua.go

```diff
@@ -0,0 +1,41 @@
+package version
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"sync"
+)
+
+// UAExtraProducts is a map of extra product information to be included in the
+// User-Agent header.
+var UAExtraProducts map[string]string
```

Comment:
I think it is better to do this with a callback. Eg. we can then set the kernel version in here that isn't completely cheap to do inside `init()`.